typescript/cli/src/config/hooks.ts

```diff
@@ -151,43 +158,52 @@ export async function createHookConfig(
   context: CommandContext,
   chain: ChainName,
   remotes: ChainName[],
+  hookTypeOverrides?: HookType[],
 ): Promise<HookConfig> {
+  let hookTypeOverride: HookType | undefined;
+  if (hookTypeOverrides) {
+    if (hookTypeOverrides.length === 1) hookTypeOverride = hookTypeOverrides[0];
+    if (hookTypeOverrides.length > 1) hookTypeOverride = HookType.AGGREGATION;
+  }
```

Comment:
instead of polluting this function with the "overrides", can we just have a default `HookConfig` that we return before entering this function?

Reply:
i'd like to use as much of the existing flow as possible, and this gives us to ability to specify HookTypes/IsmTypes directly from the CLI or config in the future

https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/0c55e1259e3b07fb7a1458623aaa1f148c79427d/typescript/cli/src/config/hooks.ts#L164-L165
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/0c55e1259e3b07fb7a1458623aaa1f148c79427d/typescript/cli/src/config/hooks.ts#L381-L389
are you alright with this instead?

Reply:
I see your comments below to pull out into an above function– going to consider this satisfactory

Reply:
> i'd like to use as much of the existing flow as possible, and this gives us to ability to specify HookTypes/IsmTypes directly from the CLI or config in the future

this seems like strictly using **less** of the existing flow because you modified this function and changed the CLI interface

Reply:
updated to this:
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/2663f26b77b373a887b478ee57e1b1a5b52820a1/typescript/cli/src/config/ism.ts#L137-L139